Documentation/network/servicemesh/mutual-authentication/mutual-authentication-example.rst

```diff
@@ -0,0 +1,235 @@
+
+    Entry ID         : 8aa91d65-16c4-48a0-bc1f-c9bf26e6a25f
+    SPIFFE ID        : spiffe://spiffe.cilium/cilium-operator
+    Parent ID        : spiffe://spiffe.cilium/ns/cilium-spire/sa/spire-agent
+    Revision         : 0
+    X509-SVID TTL    : default
+    JWT-SVID TTL     : default
+    Selector         : k8s:ns:kube-system
+    Selector         : k8s:sa:cilium-operator
+
+
+Verify that the workloads have identities registered with the SPIRE server.
+
+First, let's get the Cilium Identity of the *echo* Pod first as the SPIFFE ID - that uniquely identifies a workload - is partially based on the Cilium identity.
+
+.. code-block:: shell-session
+
+    $ IDENTITY_ID=$(kubectl get cep -l app=echo -o=jsonpath='{.items[0].status.identity.id}')
+    $ echo $IDENTITY_ID
+    17947
+
+The SPIFFE ID for this workload is based on the ``spiffe://spiffe.cilium/identity/$IDENTITY_ID`` format.
+
+Verify that this Pod has a registered identity on the SPIRE server:
+
+.. code-block:: shell-session
+
+    # kubectl exec -n cilium-spire spire-server-0 -c spire-server -- /opt/spire/bin/spire-server entry show -spiffeID spiffe://spiffe.cilium/identity/$IDENTITY_ID
+    Found 1 entry
+    Entry ID         : 9fc13971-fb19-4814-b9f0-737b30e336c6
+    SPIFFE ID        : spiffe://spiffe.cilium/identity/17947
+    Parent ID        : spiffe://spiffe.cilium/cilium-operator
+    Revision         : 0
+    X509-SVID TTL    : default
+    JWT-SVID TTL     : default
+    Selector         : cilium:mutual-auth
+
+You can see the that the *cilium-operator* was listed in the ``Parent ID``. That is because the Cilium operator creates SPIRE entries for Cilium identity, as they are created.
+
+To get all registration entries, execute the following command:
+
+.. code-block:: shell-session
+
+    kubectl exec -n cilium-spire spire-server-0 -c spire-server -- /opt/spire/bin/spire-server entry show -selector cilium:mutual-auth
+
+There are as many entries as there are identities. Verify that these match by running the command:
+
+.. code-block:: shell-session
+    
+    kubectl get ciliumidentities
+
+The identify ID listed under ``NAME`` should match with the digits at the end of the SPIFFE ID executed in the previous command.
+
+
+Enforce Mutual Authentication
+=============================
+
+Rolling out mutual authentication with Cilium is as simple as adding the following to an existing or new CiliumNetworkPolicy.
+
+Add the following to your egress or ingress rules:
+
+.. code-block:: shell-session
+
+    authentication:
+        mode: "required"
+
+Update the existing rule to only allow ingress access to mutually authenticated workloads to access *echo* using:
+
+.. parsed-literal::
+
+    $ kubectl apply -f |SCM_WEB|\/examples/kubernetes/servicemesh/cnp-with-mutual-auth.yaml
+
+Verify Mutual Authentication
+===============================
+
+Re-try your connectivity tests. They should give similar results as before. Has mutual authentication actually happened?
```

Comment:
```suggestion
Re-try your connectivity tests. They should give similar results as before:
```
We answer this question in the next step, so having it here may be confusing to users.